client/singleItemPage.jsx

```diff
@@ -8,14 +8,42 @@ import MainImageContainer from './components/mainImageContainer.jsx';
 class SingleItemPage extends React.Component{
   constructor(props){
     super(props);
-
+    this.state = {
+      image:'https://images-na.ssl-images-amazon.com/images/I/71fRayAebNL._AC_UY675_.jpg',
+      mainImages: ['https://images-na.ssl-images-amazon.com/images/I/71fRayAebNL._AC_UY675_.jpg',
+             'https://images-na.ssl-images-amazon.com/images/I/81m6GvqPiVL._AC_UY695_.jpg',
+             'https://images-na.ssl-images-amazon.com/images/I/716Htx4xkDL._AC_UY695_.jpg',
+             'https://images-na.ssl-images-amazon.com/images/I/819qaiUyqjL._AC_UX675_.jpg',
+             'https://images-na.ssl-images-amazon.com/images/I/81Mv8DR-9iL._AC_UX675_.jpg'
+            ],
+      alt:[
+        'https://images-na.ssl-images-amazon.com/images/I/41Z-gNS4B2L._AC_US40_.jpg',
+        'https://images-na.ssl-images-amazon.com/images/I/41akNdA9C7L._AC_US40_.jpg',
+        'https://images-na.ssl-images-amazon.com/images/I/418qx9runSL._AC_US40_.jpg',
+        'https://images-na.ssl-images-amazon.com/images/I/41LRFWMEXqL._AC_US40_.jpg',
+        'https://images-na.ssl-images-amazon.com/images/I/41qpQi0l-UL._AC_US40_.jpg',
+      ],
+      selected:{}
+    }
+    this.onHoverAlt = this.onHoverAlt.bind(this);
   }
 
+  onHoverAlt(e){
+    var index = e.target.getAttribute('value');
+    if(this.state.selected.classList){
+      this.state.selected.classList.remove('selected');
+    }
+    e.target.classList.add('selected');
+    this.setState({
+      image: this.state.mainImages[index],
+      selected:e.target
+    });

```

Comment:
Have you considered whether you will need asynchronous state changes? Just wondering if some of these hovers will happen fast enough that it would produce jumpy behavior. If you haven't experienced that then you can disregard, but if you have look into asynch setState here: https://reactjs.org/docs/state-and-lifecycle.html

Reply:
I will check that out if i run into that issue